File: java/src/Structural/Proxy/PotOfTea.java
package Structural.Proxy;


//PotOfTea.java - the Real Subject

public class PotOfTea implements PotOfTeaInterface {  
   public PotOfTea() {
       System.out.println("Making a pot of tea");
   }
   
   public String pourTea() {
       return "Pouring tea";
   }   
}

File: scala/src/main/scala/Structural/Proxy/PotOfTea.scala
package Structural.Proxy;


//PotOfTea - the Real Subject

class PotOfTea extends PotOfTeaInterface {  
   PotOfTea() {
       System.out.println("Making a pot of tea");
   }
   
   String pourTea() {
       return "Pouring tea";
   }   
}
